### PR TITLE
Added DateCreated and DateUpdated to form creation

### PR DESCRIPTION
**Pull request by impronunciable**
before this change the DateCreated was the default and we were displaying wrong dates to the users

### CODE REVIEW

File: internal/ask/form/form.go [hunk 1]
```diff
@@ -102,6 +102,8 @@ func Upsert(context interface{}, db *db.DB, form *Form) error {
 	if form.ID.Hex() == "" {
 		form.ID = bson.NewObjectId()
 		isNewForm = true
+		form.DateCreated = time.Now()
+		form.DateUpdated = time.Now()
 	}
```
Comment (wyattjoh) [inline internal/ask/form/form.go]: The DateUpdated should be outside that block.

Comment (impronunciable) [inline internal/ask/form/form.go]: @wyattjoh if we are updating this is being updated on UpdateStats

Comment (wyattjoh) [inline internal/ask/form/form.go]: It should still be added at that point, as it's important to keep the database as consistent as possible. If UpdateStats were to fail, then the DateUpdated would never be changed for a non-new form.

Comment (impronunciable) [inline internal/ask/form/form.go]: @wyattjoh so should I just move the DateUpdated outside the if or also remove it from UpdateStats?

Comment (wyattjoh) [inline internal/ask/form/form.go]: Just move this down a line, and leave UpdateStats untouched. It still performs an update there, but for changing form contents we are also updating the form, hence we should also update the date updated.

Comment (impronunciable) [inline internal/ask/form/form.go]: done

**Comment by jde** [comment]
We should also update DateUpdated when updates happen.  I solved the same problem here: https://github.com/coralproject/shelf/pull/214

**Comment by wyattjoh** [comment]
That got solved already as it's fired all the time being outside the if statement.

**Comment by wyattjoh** [comment]
DateUpdated should be set even when it's a new form, as it's technically updated on creation.

**Pull request merged**